### PR TITLE
Add [auth|security]:checkRights

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do?

Add [auth|security]:checkRights https://github.com/kuzzleio/sdk-javascript/issues/571

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### CODE REVIEW

File: doc/7/controllers/auth/check-rights/index.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+---
+
+```js
+checkRights(_id, body)
+```
+
+| Property | Type | Description |
+|--- |--- |--- |
+| `_id` | <pre>string</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/index.md:21]:
```suggestion
| `kuid` | <pre>string</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |
```

File: doc/7/controllers/auth/check-rights/index.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+| `body` | <pre>object</pre> | Contains a `request` |
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/index.md:22]:
```suggestion
| `requestPayload` | <pre>object</pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
```

**Review by Aschen — CHANGES_REQUESTED** [review]
I'm sorry Kuzzle documentation was not accurate, can you correct this PR with the new doc ? 
https://github.com/kuzzleio/kuzzle/pull/1912

Also can you choose better names for method arguments since they will be displayed in the editor with the jsdoc

**Review by Njuelle — APPROVED** [review]

**Review by rolljee — APPROVED** [review]

File: doc/7/controllers/auth/check-rights/snippets/check-rights.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const requestPayload = {
+  controller: 'document',
+  action: 'create',
+  index: 'nyc-open-data',
+  collection: 'yellow-taxi',
+  body: {
+    'name': 'Melis'
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/snippets/check-rights.js:7]:
```suggestion
    name: 'Melis'
```

File: doc/7/controllers/auth/check-rights/index.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+code: true
+type: page
+title: checkRights
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/index.md:4]:
```suggestion
title: checkRights
description: Checks if an API action can be executed by the current user
```

File: doc/7/controllers/security/check-rights/index.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+---
+code: true
+type: page
+title: checkRights
+---
+
+# checkRights
+
+<SinceBadge version="2.8.0"/>
+
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/index.md:10]:
```suggestion
<SinceBadge version="auto-version"/>
```

File: doc/7/controllers/security/check-rights/snippets/check-rights.test.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+name: security#checkRights
+description: Checks if an API action can be executed by the current user
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/snippets/check-rights.test.yml:2]:
```suggestion
description: Checks if an API action can be executed by a user
```

File: doc/7/controllers/security/check-rights/snippets/check-rights.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const requestPayload = {
+  controller: 'document',
+  action: 'create',
+  index: 'nyc-open-data',
+  collection: 'yellow-taxi',
+  body: {
+    'name': 'Melis'
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/snippets/check-rights.js:7]:
```suggestion
    name: 'Melis'
```

File: doc/7/controllers/security/check-rights/index.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
+---
+code: true
+type: page
+title: checkRights
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/index.md:4]:
```suggestion
title: checkRights
description: Checks if an API action can be executed by a user
```

File: doc/7/controllers/auth/check-rights/index.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+---
+
+# checkRights
+
+<SinceBadge version="Kuzzle 2.8.0"/>
+<SinceBadge version="auto-version"/>
+
+Checks if the provided API request can be executed by a user.
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/index.md:12]:
```suggestion
Checks if the provided API request can be executed by the current logged user.
```

File: doc/7/controllers/auth/check-rights/index.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+---
+
+```js
+checkRights(kuid, requestPayload)
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/index.md:17]:
```suggestion
checkRights(requestPayload)
```

File: doc/7/controllers/auth/check-rights/index.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+| Property | Type | Description |
+|--- |--- |--- |
+| `kuid` | <pre>string</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/index.md:22]:
```suggestion
|--- |--- |--- |
```

File: doc/7/controllers/auth/check-rights/index.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+| `requestPayload` | <pre>object</pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
+
+## Body properties
```
Comment (Aschen) [inline doc/7/controllers/auth/check-rights/index.md:25]:
```suggestion
## `requestPayload`
```

File: doc/7/controllers/security/check-rights/index.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+```js
+checkRights(kuid, requestPayload)
+```
+
+| Property | Type | Description |
+|--- |--- |--- |
+| `kuid` | <pre>string</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |
+| `requestPayload` | <pre>object</pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
+
+## Body properties
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/index.md:24]:
```suggestion
## `requestPayload`
```

File: doc/7/controllers/security/check-rights/index.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+The body must contain a [RequestPayload](/core/2/api/payloads/request) with at least the following properties:
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/index.md:26]:
```suggestion
The [RequestPayload](/core/2/api/payloads/request) must contains at least the following properties:
```

File: src/controllers/Auth.ts [hunk 15]
```diff
@@ -91,6 +91,27 @@ export class AuthController extends BaseController {
...
+    requestPayload: JSONObject
+  ): Promise<boolean> {
+
+    const request = {
+      _id: kuid,
+      body: requestPayload,
+      action: 'checkRights'
+    };
+    return this.query(request)
+      .then(response => response.result);
```
Comment (Aschen) [inline src/controllers/Auth.ts:112]:
```suggestion
      .then(response => response.result.allowed);
```

File: src/controllers/Security.js [hunk 16]
```diff
@@ -39,6 +39,22 @@ class SecurityController extends BaseController {
...
+   * @param {Object} body - Request to check
+   */
+  checkRights(userId, body) {
+    const request = {
+      userId,
+      body,
+      action: 'checkRights'
+    };
+    return this.query(request)
+      .then(response => response.result);
```
Comment (Aschen) [inline src/controllers/Security.js:55]:
```suggestion
      .then(response => response.result.allowed);
```

File: src/controllers/Security.js [hunk 17]
```diff
@@ -39,6 +39,22 @@ class SecurityController extends BaseController {
       .then(response => response.result);
   }
 
+  /**
+   * Checks if an API action can be executed by the current user
+   * 
+   * @param {String} userId - User kuid
+   * @param {Object} body - Request to check
+   */
+  checkRights(userId, body) {
```
Comment (Aschen) [inline src/controllers/Security.js:48]:
```suggestion
  checkRights(kuid requestPayload) {
```

File: src/controllers/Security.js [hunk 18]
```diff
@@ -39,6 +39,22 @@ class SecurityController extends BaseController {
...
+    const request = {
+      userId,
```
Comment (Aschen) [inline src/controllers/Security.js:50]:
```suggestion
      userId: kuid,
```

File: src/controllers/Security.js [hunk 19]
```diff
@@ -39,6 +39,22 @@ class SecurityController extends BaseController {
...
+      body,
```
Comment (Aschen) [inline src/controllers/Security.js:51]:
```suggestion
      body: requestPayload,
```

File: doc/7/controllers/security/check-rights/snippets/check-rights.js [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+  }
+}
+
+try {
+  const result = await kuzzle.security.checkRights('foo', requestPayload);
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/snippets/check-rights.js:12]:
```suggestion
  const allowed = await kuzzle.security.checkRights('foo', requestPayload);
```

File: doc/7/controllers/security/check-rights/snippets/check-rights.js [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+  console.log(result.allowed);
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/snippets/check-rights.js:13]:
```suggestion
  console.log(allowed);
```

File: doc/7/controllers/security/check-rights/index.md [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+
+- `controller`: API controller
+- `action`: API action
+
+---
+
+## Resolves
+
+The returned result contains the following property:
```
Comment (Aschen) [inline doc/7/controllers/security/check-rights/index.md:35]: The `allowed` boolean is directly returned

**Review by Aschen — CHANGES_REQUESTED** [review]
The methods should directly returns the boolean IMHO

**Review by rolljee — APPROVED** [review]

**Review by Njuelle — APPROVED** [review]

File: src/controllers/Auth.ts [hunk 23]
```diff
@@ -91,6 +91,24 @@ export class AuthController extends BaseController {
...
+  checkRights (
+    requestPayload: JSONObject
+  ): Promise<boolean> {
+
+    const request = {
+      body: requestPayload,
+      action: 'checkRights'
+    };
+    return this.query(request)
+      .then(response => response.result);
```
Comment (Aschen) [inline src/controllers/Auth.ts:109]:
```suggestion
      .then(response => response.result.allowed);
```

**Review by Aschen — APPROVED** [review]

**Pull request merged**